### ISSUE TITLE
Click randomly fails

### ISSUE THREAD

**Issue by yecine06** [Codeception/Codeception#1608]
hello,

I'm having an issue with the click. Without changing anything to my code sometimes the test passes and other it fails.

This is random i'm not sure why it's not working

I'm using laravel 4 and codeception 2.0.*-dev

composer file
````

"require": {
        "laravel/framework": "4.2.*",
        "maatwebsite/excel": "1.*",
        "creitive/breadcrumbs": "dev-master",
        "anahkiasen/former": "dev-master",
        "doctrine/dbal": "2.5.*@dev"
    },
    "require-dev": {
        "dollar/generators": "dev-master",
        "codeception/codeception": "2.0.*@dev",
        "laracasts/testdummy": "~2.0",
        "fzaninotto/faker": "1.4.*"
    },
````

Config yml file
````
# Codeception Test Suite Configuration

# suite for functional (integration) tests.
# emulate web requests and make application process them.
# Include one of framework modules (Symfony2, Yii2, Laravel4) to use it.

class_name: FunctionalTester
modules:
    enabled: [Filesystem, FunctionalHelper, Laravel4, Asserts]

````

Also here is my cept file

````
$I = new FunctionalTester($scenario);
$I->am('a admin');
$I->wantTo('sign in into my account');

$email = 'admin@example.com';
$password = 'secret';
$passwordHashed = Hash::make($password);
$I->haveAccount(['email' => $email, 'password' => $passwordHashed]);


$I->amOnPage('/login');
$I->fillField('email',$email);
$I->fillField('password',$password);
$I->click('Log in');

$I->seeInCurrentUrl('/pos');
````

and here are some log from the command 

````
agrant@homestead:~/Sites/bo$ vendor/bin/codecept run --debug -vvv
Codeception PHP Testing Framework v2.0.9
Powered by PHPUnit 4.4.0 by Sebastian Bergmann.

Acceptance Tests (0) ------------------------
Modules: PhpBrowser, AcceptanceHelper
---------------------------------------------
---------------------------------------------

Functional Tests (1) -------------------------------------------------------------------------------------
Modules: Filesystem, FunctionalHelper, Laravel4
----------------------------------------------------------------------------------------------------------
Trying to sign in into my account (SignInCept)
Scenario:
* As a a admin
* I have an account {"email":"admin@example.com","password":"$2y$10$r1EvmYwptiBIVMpC6mHhbeUNddlpgiXeNn0mg.T4q6mSdLacd3l46"}
* I am on page "/login"
  [Response] 200
  [Page] http://localhost/login
  [Cookies] []
  [Headers] {"cache-control":["no-cache"],"date":["Thu, 25 Dec 2014 01:11:25 GMT"],"content-type":["text/html; charset=UTF-8"]}
* I fill field "email","admin@example.com"
* I fill field "password","secret"
* I click "Log in"
  [Uri] http://localhost/login
  [POST] {"email":"admin@example.com","password":"secret","remember":"on"}
 ERROR

----------------------------------------------------------------------------------------------------------

Unit Tests (0) ------------------------------
Modules: Asserts, UnitHelper
---------------------------------------------
---------------------------------------------


Time: 4.38 seconds, Memory: 31.00Mb

There was 1 error:

---------
1) Failed to sign in into my account in SignInCept (/home/vagrant/Sites/bo/tests/functional/SignInCept.php)
Couldn't click "Log in":
ErrorException: Trying to get property of non-object

Scenario Steps:
6. I click "Log in"
5. I fill field "password","secret"
4. I fill field "email","admin@example.com"
3. I am on page "/login"
2. I have an account {"email":"admin@example.com","password":"$2y$10$r1EvmYwptiBIVMpC6mHhbeUNddlpgiXeNn0mg.T4q6mSdLacd3l46"}
1. As a a admin


  [PHPUnit_Framework_ExceptionWrapper] Trying to get property of non-object


FAILURES!
Tests: 1, Assertions: 0, Errors: 1.
vagrant@homestead:~/Sites/bo$ vendor/bin/codecept run --debug -vvv
Codeception PHP Testing Framework v2.0.9
Powered by PHPUnit 4.4.0 by Sebastian Bergmann.

Acceptance Tests (0) ------------------------
Modules: PhpBrowser, AcceptanceHelper
---------------------------------------------
---------------------------------------------

Functional Tests (1) -------------------------------------------------------------------------------------
Modules: Filesystem, FunctionalHelper, Laravel4
----------------------------------------------------------------------------------------------------------
Trying to sign in into my account (SignInCept)
Scenario:
* As a a admin
* I have an account {"email":"admin@example.com","password":"$2y$10$V6sGR3n0sBf3VZcH7zGPEuNIGI2KxvhInMjWjt/TPlGQbkYu.5aTa"}
* I am on page "/login"
  [Response] 200
  [Page] http://localhost/login
  [Cookies] []
  [Headers] {"cache-control":["no-cache"],"date":["Thu, 25 Dec 2014 01:12:28 GMT"],"content-type":["text/html; charset=UTF-8"]}
* I fill field "email","admin@example.com"
* I fill field "password","secret"
* I click "Log in"
  [Uri] http://localhost/login
  [POST] {"email":"admin@example.com","password":"secret","remember":"on"}
  [Response] 200
  [Page] http://localhost/pos
  [Cookies] []
  [Headers] {"cache-control":["no-cache"],"date":["Thu, 25 Dec 2014 01:12:29 GMT"],"content-type":["text/html; charset=UTF-8"]}
* I see in current url "/pos"
 PASSED

----------------------------------------------------------------------------------------------------------

Unit Tests (0) ------------------------------
Modules: Asserts, UnitHelper
---------------------------------------------
---------------------------------------------


Time: 4.61 seconds, Memory: 31.25Mb

OK (1 test, 1 assertion)
````

**Comment by caffeinelogic** [comment]
Did you resolve your issue? I am having a similar one where the first click works fine but then the second one does not on the same page.

**Comment by yecine06** [comment]
Nope sorry i haven't found a solution sorry

**Comment by caffeinelogic** [comment]
I was able to work around this by switching to using Selenium. My tests are more consistent and it wasn't very difficult to start using.

**Comment by janhenkgerritsen** [comment]
@caffeinelogic If switching to Selenium resolved the issue chances are that your code relied on Javascript to function correctly, which is not supported by tests run with PhpBrowser/InnerBrowser.

@yecine06 can you tell me if this issue still exists with the latest version of the Laravel 4 module? I just merged a rewrite of the module into the 2.0 branch: 132f3af48550910338fee08fedb1a078724cc2d6. This rewrite should be available when Codeception 2.0.14 is released, but you can test it already by using `"Codeception/Codeception": "2.0.*@dev"` in your `composer.json` file.

**Comment by janhenkgerritsen** [comment]
Closing this issue because I am cleaning up the open Laravel issues. If this is still an issue you can reopen it again.
